src/utils/work/fetchWork.js

```diff
@@ -139,30 +141,30 @@ const fetchWork = async (
   // Then, we may be able to find this in S3.
   let response = await seekFromS3(ETag, experimentId);
 
-  // If response cannot be fetched, go to the worker.
+  // If there is no response in S3, dispatch workRequest via the worker
   if (!response) {
-    response = await seekFromAPI(
+    await dispatchWorkRequest(
       experimentId,
       body,
       timeout,
       ETag,
-      eventCallback,
       {
         PipelineRunETag: qcPipelineStartDate,
         ...extras,
       },
     );
   }
 
-  if (!response) {
-    console.debug(`No response immediately resolved for ${body} (ETag: ${ETag}) -- this is probably an event subscription.`);
-    return response;
-  }
+  response = await seekFromS3(ETag, experimentId);
```

Comment:
In `seekWorkResponse` in the deleted code, we used to also resolve or reject the promise like this after calling seekFromS3:
<img width="507" alt="Screenshot 2022-01-31 at 13 47 30" src="https://user-images.githubusercontent.com/65414651/151796366-774559ba-bf7f-4dc6-9501-c2f5684c5bc5.png">

Why is this not necessary anymore?

Reply:
This `resolve`/`reject` in this case is to return the result of fetching the data from S3 (`resolve` if success, `reject` if error). This is not necessary anymore because the call to S3 is moved out from `workflowDispatch` and is placed afterwards.